metacat-common/src/main/java/com/netflix/metacat/common/QualifiedName.java

```diff
@@ -49,6 +49,9 @@
     private Map<String, String> parts;
     private Type type;
 
+    // Null if this QualifiedName was all lower-case to begin with.
+    private QualifiedName caseSensitiveQualifiedName;
```

Comment:
What is the rationale of not using Optional?

Reply:
Optional is not serializable and also seems to be a general recommendation that Optional should be used in return values and not for class attributes: https://stackoverflow.com/questions/29033518/is-it-a-good-practice-to-use-optional-as-an-attribute-in-a-class